src/market_mad/remotes/linuxcontainers/monitor

```diff
@@ -50,10 +58,9 @@ class LinuxContainersMarket
         @options.merge!(options)
 
         version_path = File.dirname(__FILE__) + '/../../VERSION'
+        value = "OpenNebula #{File.read(version_path)}"
 
-        if File.exists? version_path
-            @options[:agent] = "OpenNebula #{File.read(version_path)}"
-        end
+        @options[:agent] = value if File.exist? version_path
```

Comment:
File.read needs to happen only if File.exists so this should be changed to:

@options[:agent] = "OpenNebula #{File.read(version_path)}" if File.exist? version_path

Otherwise the first File.read may throw an exception